packages/ods/src/components/pagination/src/components/ods-pagination/ods-pagination.tsx

```diff
@@ -0,0 +1,329 @@
+
+      this.odsPaginationItemPerPageChanged.emit({
+        current: this.itemPerPage,
+        currentPage: this.current,
+        totalPages: this.actualTotalPages,
+      });
+    }
+  }
+
+  @Watch('totalItems')
+  async onTotalItemsChange(): Promise<void> {
+    await this.updatePagination();
+  }
+
+  @Method()
+  async setCurrentPage(current: number): Promise<void> {
+    this.current = current;
+  }
+
+  @Method()
+  async getCurrentPage(): Promise<number> {
+    return this.current;
+  }
+
+  componentWillLoad(): void {
+    this.hostId = this.el.id || getRandomHTMLId();
+
+    this.itemPerPage = ODS_PAGINATION_PER_PAGE_OPTIONS.includes(this.defaultItemsPerPage) && this.defaultItemsPerPage || ODS_PAGINATION_PER_PAGE.option_10;
+
+    if (this.totalItems) {
+      this.actualTotalPages = computeActualTotalPages(this.itemPerPage, this.totalItems, this.totalPages);
+    } else {
+      this.actualTotalPages = this.totalPages;
+    }
+
+    this.current = getActualPage(this.defaultCurrentPage, this.actualTotalPages, this.current);
+
+    this.updatePageList();
+    this.isFirstLoad = false;
+  }
+
+  private updatePageList(): void {
+    this.pageList = createPageList(this.actualTotalPages, this.current);
+  }
+
+  private emitChange(current: number, oldCurrent?: number): void {
+    this.odsPaginationChanged.emit({
+      current: current,
+      itemPerPage: this.itemPerPage,
+      oldCurrent: oldCurrent,
+    });
+  }
+
+  private async updatePagination(): Promise<void> {
+    this.actualTotalPages = computeActualTotalPages(this.itemPerPage, this.totalItems, this.totalPages);
+
+    if (this.current === 1) {
+      // If current is already 1 we don't want to emit a change event
+      this.updatePageList();
+    } else {
+      await this.setCurrentPage(1);
+    }
+  }
+
+  handlePreviousClick(page: number): void {
+    this.setCurrentPage(page - 1);
+  }
+
+  handleNextClick(page: number): void {
+    this.setCurrentPage(page + 1);
+  }
+
+  handlePageClick(page: number): void {
+    this.setCurrentPage(page);
+  }
+
+  handlePreviousKeyUp(event: KeyboardEvent, page: number): void {
+    if (this.current > 1) {
+      this.onKeyUp(event, page - 1);
+    }
+  }
+
+  handleNextKeyUp(event: KeyboardEvent, page: number): void {
+    if (this.current < this.pageList.length) {
+      this.onKeyUp(event, page + 1);
+    }
+  }
+
+  handlePageKeyUp(event: KeyboardEvent, page: number): void {
+    this.onKeyUp(event, page);
+  }
+
+  onKeyUp(event: KeyboardEvent, page: number): void {
+    if (event.key === ' ' || event.key === 'Enter') {
+      event.preventDefault();
+      this.setCurrentPage(page);
+    }
+  }
+
+  renderArrow(direction: 'left' | 'right'): typeof Fragment {
+    const isLeft = direction === 'left';
+    const tooltipLabel = isLeft ? this.labelTooltipPrevious : this.labelTooltipNext;
+    const arrowButtonId = isLeft ? this.leftArrowButtonId : this.rightArrowButtonId;
+
+    const checkArrowDisabled = (): boolean => {
```

Comment:
why a function?
it will be called multiple times on each render for the same result each time